### ISSUE TITLE
Should the date_index be lagged? 

### ISSUE THREAD

**Issue by freddyaboulton** [alteryx/evalml#2959]
This came up during my design for #2511

## Problem
Currently, time series pipelines created by AutoMLSearch will lag the features created from the date time index. This will create a lot of redundant/highly correlated features. 

Repro 
```python
from evalml.automl import AutoMLSearch
from evalml.demos import load_weather

X, y = load_weather()

automl = AutoMLSearch(X, y, problem_type="time series regression",
                      max_batches=1,
                      problem_configuration={"gap": 0, "max_delay": 7,
                                             "forecast_horizon": 7, "date_index": "Date"},
                      allowed_model_families=["random_forest"],
                      verbose=True)
automl.search()
automl.best_pipeline.transform_all_but_final(X, y)
```

You should see that the pipeline creates 40 features

#### Current pipeline features
![image.png](https://images.zenhubusercontent.com/5ef0fc259b404dbc3f292b37/0bf6a75e-b404-449e-a28d-0bf9fef09814)

### Explanation
What's happening is that the pipeline turns the date_index column into features via the `DateTimeFeaturizer` then the `DelayedFeatureTransformer` computes lags for these features:
![image.png](https://images.zenhubusercontent.com/5ef0fc259b404dbc3f292b37/bddc6ec7-db63-46c9-b851-3d96819e9d68)

### One possible solution
I wonder if it's better to treat the `date_index` as a Known-in-advanced feature. If we follow the design in #2511, this will make sure the date_index is not lagged. And it would mean the pipeline features would look something like

#### Desired pipeline features
![image.png](https://images.zenhubusercontent.com/5ef0fc259b404dbc3f292b37/a9b69f2f-8e42-4cd5-ac7a-46acfc64aa7c)

which I think is better.

I like the idea of treating the date index as known-in-advanced because it's known-in-advanced! But there could be other solutions for solving the problem as stated and achieving the desired pipeline features.

**Comment by dsherry** [comment]
@freddyaboulton I agree with your assessment! The date indicators we currently create will always be known in advance. I could see someone wanting to add custom date indicators for unknown events (`is_covid_a_problem_today`), but that seems like a future challenge.

So, the main fix for this issue will be tracked by the known-in-advance work. Is there anything we need to do in the short term, like prevent the date index from being used to create lagged features?

**Comment by freddyaboulton** [comment]
@dsherry Not lagging the date index will happen in https://github.com/alteryx/evalml/issues/3031 !

**Comment by freddyaboulton** [comment]
Closed by #3031
